src/apps/allbridge/positions.ts

```diff
@@ -36,32 +40,53 @@ const hook: PositionsHook = {
 
     const client = getClient(networkId)
 
-    const [balances, rewards] = await Promise.all([
-      await Promise.all(
-        allbridgeTokenInfo.map(async (tokenInfo) => {
-          return address
-            ? client.readContract({
-                address: tokenInfo.poolAddress,
-                abi: poolAbi,
-                functionName: 'balanceOf',
-                args: [address as Address],
-              })
-            : undefined
-        }),
-      ),
-      await Promise.all(
-        allbridgeTokenInfo.map(async (tokenInfo) => {
-          return address
-            ? client.readContract({
-                address: tokenInfo.poolAddress,
-                abi: poolAbi,
-                functionName: 'pendingReward',
-                args: [address as Address],
-              })
-            : undefined
-        }),
-      ),
-    ])
+    const [balances, rewards, totalSupplies, lpTokenDecimals] =
+      await Promise.all([
```

Comment:
Same question here of awaiting within a promise.all.

Reply:
updated in [6bdca0f](https://github.com/valora-inc/hooks/pull/499/commits/6bdca0f6c61827cdc43bf9f5d44274280e1e3a00)